src/document/json/tree.ts

```diff
@@ -133,6 +134,41 @@ function createCRDTTreeNode(context: ChangeContext, content: TreeNode) {
   return root;
 }
 
+/**
+ * `validateTextNode` ensures that a text node has a non-empty string value.
+ */
+function validateTextNode(textNode: TextNode): boolean {
+  if (!textNode.value.length) {
+    throw new Error('text node cannot have empty value');
+  } else {
+    return true;
+  }
+}
+
+/**
+ * `validateTreeNodes` ensures that treeNodes consists of only one type.
+ */
+function validateTreeNodes(treeNodes: Array<TreeNode>): boolean {
+  const firstTreeNodeType = treeNodes[0].type;
+  if (firstTreeNodeType === 'text') {
```

Comment:
You can change string `text` to `DefaultTextType` by importing it from index_tree.ts